### PR TITLE
Implement Progress events to help diagnose RefreshPorts & Tunnel Service Requests 

### PR DESCRIPTION
**Pull request by jramsay**
Implement Progress events to help diagnose RefreshPorts & Tunnel Service Requests

### CODE REVIEW

**Comment by jramsay** [comment]
Note: this PR will need to be updated with the new SSH version before it can be completed

https://github.com/microsoft/dev-tunnels-ssh/pull/92

File: cs/src/Management/TunnelManagementClient.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@
 using System.Threading.Tasks;
 using System.Web;
 using Microsoft.DevTunnels.Contracts;
+using Microsoft.DevTunnels.Ssh;
```
Comment (jramsay) [inline cs/src/Management/TunnelManagementClient.cs:21]: Note: these errors will be resolved when I update to use the SSH package. In my local testing I've included a link to  the project reference. Requires the companion SSH PR to be checked in first.

File: cs/src/Management/TunnelManagementClient.cs [hunk 2]
```diff
@@ -1337,6 +1352,15 @@ public void Dispose()
             return result;
         }
 
+        /// <summary>
+        /// Event fired when a progress connection event has been reported.
+        /// </summary>
+        private void OnReportProgress(Progress progress)
+        {
+            var args = new SshReportProgressEventArgs(progress);
```
Comment (IlyaBiryukov) [inline cs/src/Management/TunnelManagementClient.cs:1360]: Nit: you can save on args object allocation if you do it inside if:
``` csharp
if (ReportProgress is EventHandler<SshReportProgressEventArgs> handler)
{
    var args = new SshReportProgressEventArgs(progress);
    handler.Invoke(this, args);
}
```

File: cs/src/Management/TunnelManagementClient.cs [hunk 3]
```diff
@@ -1337,6 +1352,15 @@ public void Dispose()
             return result;
         }
 
+        /// <summary>
+        /// Event fired when a progress connection event has been reported.
+        /// </summary>
+        private void OnReportProgress(Progress progress)
```
Comment (IlyaBiryukov) [inline cs/src/Management/TunnelManagementClient.cs:1358]: Nit: making it `protected virtual` will let derived classes handle progress report without the need of subscribing to the event.

**Review by IlyaBiryukov — DISMISSED** [review]

**Review by jasongin — APPROVED** [review]

**Pull request merged**